src/thor/costmatrix.cc

```diff
@@ -516,12 +547,23 @@ bool CostMatrix::ExpandInner(baldr::GraphReader& graphreader,
   bool not_thru_pruning =
       not_thru_pruning_ ? (pred.not_thru_pruning() || !meta.edge->not_thru()) : false;
 
+  float dist = 0.0f;
+  auto newsortcost =
+      newcost.cost +
+      GetAstarHeuristic<expansion_direction>(index, t2->get_node_ll(meta.edge->endnode()), dist);
+
   // Add edge label, add to the adjacency list and set edge status
   uint32_t idx = edgelabels.size();
   *meta.edge_status = {EdgeSet::kTemporary, idx};
   if (FORWARD) {
-    edgelabels.emplace_back(pred_idx, meta.edge_id, opp_edge_id, meta.edge, newcost, mode_, tc,
-                            pred_dist, not_thru_pruning,
+    if (hierarchy_limits_[MATRIX_FORW][index][meta.edge_id.level()].max_up_transitions !=
+        kUnlimitedTransitions) {
+      // Override distance to the destination with a distance from the origin.
+      // It will be used by hierarchy limits
+      dist = astar_heuristics_[MATRIX_REV][index].GetDistance(t2->get_node_ll(meta.edge->endnode()));
+    }
```

Comment:
same logic as in bidirectional_astar.cc: don't need the heuristic distance if we're not having any hierarchy limits. though I wonder if that couldn't be made a constant for each request; AFAIK we never change to `kUnlimitedTransitions` dynamically..